### ISSUE TITLE
Problem with knobs acceleration

### ISSUE THREAD

**Issue by mystickaleph** [raphaelquast/beatstep#78]
How to control knob sensitivity?

**Comment by raphaelquast** [comment]
whats the problem? 
if the knobs are somewhat jumpy that's most probably the controller and not the script...

**Comment by mystickaleph** [comment]
The fact is that knobs 8 and 16 in the factory script have a normal ratio (1:1), the rest are 1:2. As for your script, here all knobs have a ratio of 1:3. Is it possible to add an option with which you can control these parameters manually?

**Comment by raphaelquast** [comment]
well... in theory it's of course possible... 
in the script, the sensitivity of the encoders is not the same for all layers... 
(for example scrolling tracks & scenes has a different sensitivity than changing device parameters)
what specific functionality are you talking about? the device controls?

**Comment by mystickaleph** [comment]
First of all, device controls, but ideally it would be great to control the ratio of each encoder. The problem of "slow" encoders on beatstep is quite old, and no one has solved the problem properly for 7 years. Here is a discussion on the official Arturia forum: https://forum.arturia.com/index.php?topic=82122.60

**Comment by raphaelquast** [comment]
yep I know... the encoders a pain...  in my case not because they are slow, but because (at least on my controller) they also tend to be "jumpy"

The problem with implementing a encoder-sensitivity funtion is that the device controls in the BeatstepQ script are not yet custom implemented, but they use Ableton's Framework-classes, e.g.:
https://github.com/gluon/AbletonLive10.1_MIDIRemoteScripts/blob/master/_Framework/DeviceComponent.py

so one would need to temper with that class to incorporate some kind of sensitivity adjustment (or re-implement a small-scale version of it that does the basic things + sensitivity)
unfortunately (as basically all of Ableton's python API) it's nearly completely undocumented and so this most probably requires some work to actually get it done... I am currently a bit short on time so it'll take a while until I find some time to look at this...
if you know a bit of python, give it a try!   (otherwise I can put it on my to-do list but no guarantees that it'll be solved soon 😄 )

**Comment by ifesdjeen** [comment]
My guess that the problem might be not exactly knob acceleration, but rather their mode. Try re-flashing the firmware with settings that would switch encoders to Relative mode.
